### PR TITLE
Fix path resolution in Pages esbuild

### PR DESCRIPTION
**Pull request by JacobMGEvans**
fix: the pathing for the template was resolving using `__dirname` into the esbuild file which then used a relative path back from pages. When ran inside a test would get a resolved absolute path to the template from "pages/pages/functions/template-workers.ts

### CODE REVIEW

**Review by petebacondarwin — APPROVED** [review]
Let's wait for @GregBrimble to confirm this is a bug before we merge.

**Comment by petebacondarwin** [comment]
Oh interesting...

https://github.com/cloudflare/wrangler2/runs/5892721660?check_suite_focus=true#step:6:942

This implies that the original pathing is correct, at least for how that e2e test is setup.

File: packages/wrangler/pages/functions/buildWorker.ts
Comment (petebacondarwin) [inline packages/wrangler/pages/functions/buildWorker.ts:24]: I wonder if `__dirname` returns a different result once this file is bundled?

In other words, the `__dirname` at runtime is `packages/wrangler/wrangler-dist` rather than the source path of `packages/wrangler/pages/functions`.

I note that this only came up when we were trying to run Pages in a Wrangler unit test, which is not bundled...

**Comment by threepointone** [comment]
Tests are failing too, expected?

**Comment by petebacondarwin** [comment]
So the problem is that esbuild does not rewrite `__dirname` when bundling. So it refers to the directory containing the bundle, rather than the directory containing the original source file.
This is problematic since you do not necessarily know where the final bundle will be run compared to the file that is being loaded.
Apparently there there is an esbuild plugin to solve this.

**Comment by JacobMGEvans** [comment]
> So the problem is that esbuild does not rewrite `__dirname` when bundling. So it refers to the directory containing the bundle, rather than the directory containing the original source file. This is problematic since you do not necessarily know where the final bundle will be run compared to the file that is being loaded. Apparently there there is an esbuild plugin to solve this.

I will take a look into implementing the plugin in this PR then we can discuss adding it to ESBuild for the Wrangler side as well.

**Comment by threepointone** [comment]
I'm curious too bout what @GregBrimble asked; could you explain exactly what this fixes? I don't think I understand

**Comment by petebacondarwin** [comment]
https://github.com/evanw/esbuild/issues/859

**Comment by petebacondarwin** [comment]
Any update on this @JacobMGEvans ?

**Comment by JacobMGEvans** [comment]
> Any update on this @JacobMGEvans ?

Not at the moment.

**Comment by JacobMGEvans** [comment]
Closing due to work in #860

**Pull request closed without merging**
